src/storage/txn/process.rs

```diff
@@ -170,14 +175,17 @@ impl<E: Engine, S: MsgScheduler> Executor<E, S> {
                     .inc();
 
                 error!("get snapshot failed"; "cid" => task.cid, "err" => ?err);
-                notify_scheduler(
-                    self.take_scheduler(),
-                    Msg::FinishedWithErr {
-                        cid: task.cid,
-                        err: Error::from(err),
-                        tag: task.tag,
-                    },
-                );
+                self.take_pool().pool.spawn(move || {
```

Comment:
The `execute` function must be running in scheduler worker? so we don't spawn here?